soc/arm/st_stm32/stm32l1/soc.h

```diff
@@ -51,6 +51,12 @@
 #include <stm32l1xx_ll_i2c.h>
 #endif
 
+#if defined(CONFIG_RTC_STM32) || defined(CONFIG_COUNTER_RTC_STM32)
```

Comment:
CONFIG_RTC_STM32 is no more supported, please remove

Reply:
Could `drivers/counter/Kconfig.stm32_rtc` be removed too?

Reply:
Actually counter_rtc replaced rtc. rtc has been removed, but counter_rtc is still needed.